extensions/wikia/UserLogin/UserLoginHelper.class.php

```diff
@@ -624,9 +624,12 @@ public function getNewAuthUrl( $page = '/join' ) {
 			$requestUrl = $this->app->wg->request->getRequestURL();
 		}
 
-		return $page . '?redirect='
+		parse_str( parse_url( $page, PHP_URL_QUERY ), $queryParams );
+
+		return wfAppendQuery( $page, 'redirect='
 			. urlencode ( wfExpandUrl ( $requestUrl ) )
-			. $this->getUselangParam();
+			. ( !isset( $queryParams['uselang'] ) ? $this->getUselangParam() : '' )
```

Comment:
Done in: https://github.com/Wikia/app/pull/11379/commits/cc57b05bf25adc8267b85738fc545a459c012220